apps/core/views.py

```diff
@@ -143,3 +151,145 @@ class DecoratedTokenObtainPairView(TokenObtainPairView):
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        return Response(
+            {"detail": "Reset password email sent"},
+            status=status.HTTP_201_CREATED,
+        )
+
+
+class PasswordResetConfirmView(APIView):
+    permission_classes = (AllowAny,)
+
+    @swagger_auto_schema(
+        tags=['auth'],
+        manual_parameters=[
+            openapi.Parameter(
+                'user_id', openapi.IN_QUERY,
+                description="user unique id",
+                type=openapi.TYPE_STRING
+            ),
+            openapi.Parameter(
+                'confirmation_token',
+                openapi.IN_QUERY,
+                description="confirmation token",
+                type=openapi.TYPE_STRING
+            )
+        ],
+    )
+    def get(self, request, *args, **kwargs):
+        user_id = request.query_params.get('user_id', '')
+        confirmation_token = request.query_params.get('confirmation_token', '')
+        user_model = get_user_model()
+
+        try:
+            user = user_model.objects.get(id=user_id)
+        except(TypeError, ValueError, OverflowError, User.DoesNotExist):
+            user = None
+
+        if not User:
+            return Response(
+                'User not found',
+                status=status.HTTP_404_NOT_FOUND
+            )
+
+        if not default_token_generator.check_token(user, confirmation_token):
+            return Response(
+                data='Token is invalid or expired. '
+                     'Please request another password changing.',
+                status=status.HTTP_401_UNAUTHORIZED,
+            )
+
+        reset_token = default_token_generator.make_token(user)
+        return Response({"reset_token": reset_token})
+
+
+class PasswordChangeView(APIView):
+    queryset = User.objects.all()
+    permission_classes = (AllowAny,)
+    serializer_class = PasswordChangeSerializer
+
+    def get_serializer(self, *args, **kwargs):
+        return self.serializer_class(self, *args, **kwargs)
+
+    @swagger_auto_schema(
+        tags=['auth'],
+        manual_parameters=[
+            openapi.Parameter(
+                'user_id',
+                openapi.IN_QUERY,
+                description="user unique id",
+                type=openapi.TYPE_STRING,
+            ),
+            openapi.Parameter(
+                'reset_token',
+                openapi.IN_QUERY,
+                description="reset token",
+                type=openapi.TYPE_STRING,
+            )
+        ],
+    )
+    def post(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        user_id = request.query_params.get('user_id', '')
+        reset_token = request.query_params.get('reset_token', '')
+        user_model = get_user_model()
+
+        try:
+            user = user_model.objects.get(id=user_id)
+        except(TypeError, ValueError, OverflowError, User.DoesNotExist):
+            user = None
```

Comment:
Let's add logging to understand what's going on in the backend

Reply:
It will be enough just to specify the error, the error level (warning) and the name of the file where it happened will be added by the logger to the message.
` logger.warning(f'Get user feialed: {exc}')`